### PR TITLE
Add type validation to mock constructor

### LINKED ISSUE #142: Update documentation
- `StrictMock`
  - `type_validation` attribute (at `__init__` and `docs/`)
- `mock_callable`, `mock_async_callable`, `mock_constructor`:
  - - `type_validation` attribute.

### PR DESCRIPTION
**Pull request by fornellas**
Ditto.

This is currently failing due to #145, tests will likely pass once that's fixed.

Closes #146.
Closes #142.

### CODE REVIEW

**Comment by fornellas** [comment]
@david-caro  your #149  fix worked, thanks!

File: docs/patching/mock_constructor/index.rst [hunk 1]
```diff
@@ -56,6 +56,31 @@ Note how by using ``mock_constructor()``, not only you get all **safe by default
...
+  class Messenger:
+      def __init__(self, message: str):
+        self.message = message
+  
+  class TestArgumentTypeValidation(testslide.TestCase):
+      def test_argument_type_validation(self):
+          messenger_mock = testslide.StrictMock(template=Messenger)
+          self.mock_constructor(sys.modules[__name__], "Messenger").to_return_value(messenger_mock)
+          with self.assertRaises(TypeError):
+            # TypeError: Call with incompatible argument types:
```
Comment (david-caro) [inline docs/patching/mock_constructor/index.rst:78]: nit: indents are weird (2 spaces here, 4 in the block before)

File: testslide/mock_constructor.py [hunk 2]
```diff
@@ -157,11 +157,24 @@ def __delete__(self, instance):
...
+    return callable_mock_with_type_validation
+
+
+def _get_mocked_class(original_class, target_class_id, callable_mock, type_validation):
     if target_class_id in _target_class_id_by_original_class_id:
         raise RuntimeError("Can not mock the same class at two different modules!")
     else:
         _target_class_id_by_original_class_id[id(original_class)] = target_class_id
+
+    original_class_new = original_class.__new__
```
Comment (david-caro) [inline testslide/mock_constructor.py:175]: I would have forgotten about new xd

Comment (fornellas) [reply testslide/mock_constructor.py:175]: Turns out, it is there only for being future proof, as `mock_constructor()` currently can't handle classes that define their own `__new__`: #155.

File: docs/patching/mock_constructor/index.rst (hunk 1 above)
Comment (fornellas) [reply docs/patching/mock_constructor/index.rst:78]: Fixed this on master after the merge, tkz.

**Pull request merged**